docs/pages/4.advanced.md

```diff
@@ -54,3 +52,44 @@ Finally, let's use it in our page.vue with vue-instantsearch components:
   </div>
 </template>
 ```
+
+## TypeScript
+
+To allow for index names auto-completion and search results typings, you may define your indices expected results by extending the `AlgoliaIndices` interface as follows:
+
+```ts
+// types.d.ts
+import '@nuxtjs/algolia'
+
+declare module '@nuxtjs/algolia' {
+  interface AlgoliaIndices {
+    someIndex: {
+      foo: string;
+      bar: number;
+    }
+  }
+}
+```
+
+You may then use your Algolia indices and enjoy typed search hits 🔥
+
+```vue
+<script lang="ts" setup>
+const { search } = useInitIndex('someIndex') // Auto-completion for index names
```

Comment:
Ok, I see what you did here :D 

useInitIndex should be used mainly to have a reference to the index itself (similarly to the useAlgolia, we use it to have a reference to algoliasearch). 

In this case I would only leave the example with the useSearch as this should be used in order to handle search functionality using Algolia.

So something like this:

```ts
const { result, search } = useSearch('someIndex')

await search({ query: 'test'  }) 
```

useSearch uses useInitIndex under the hood btw.